osu.Game/Screens/OnlinePlay/Lounge/Components/RoomSpecialCategoryPill.cs

```diff
@@ -5,58 +5,27 @@
 
 using osu.Framework.Allocation;
 using osu.Framework.Extensions;
-using osu.Framework.Graphics;
-using osu.Framework.Graphics.Sprites;
 using osu.Game.Graphics;
-using osu.Game.Graphics.Sprites;
 using osuTK.Graphics;
 
 namespace osu.Game.Screens.OnlinePlay.Lounge.Components
 {
-    public partial class RoomSpecialCategoryPill : OnlinePlayComposite
+    public partial class RoomSpecialCategoryPill : OnlinePlayPill
     {
-        private SpriteText text;
-        private PillContainer pill;
-
         [Resolved]
         private OsuColour colours { get; set; }
 
-        public RoomSpecialCategoryPill()
-        {
-            AutoSizeAxes = Axes.Both;
-        }
-
-        [BackgroundDependencyLoader]
-        private void load()
-        {
-            InternalChild = pill = new PillContainer
-            {
-                Background =
-                {
-                    Colour = colours.Pink,
-                    Alpha = 1
-                },
-                Child = text = new OsuSpriteText
-                {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    Font = OsuFont.GetFont(weight: FontWeight.SemiBold, size: 12),
```

Comment:
> To tell the truth I've grown a little allergic of applying inheritance eagerly in drawables to "reduce code duplication"

Agree and I almost closed this PR as a result. But this case does seem like it's worthwhile combining for the time being to have conformity across the pill displays.

The discrepancy you found probably means one of the two font weight choices was actually incorrect. I'd stick with what this PR uses, or adjust the base font weight to `Medium` instead (both look good enough, maybe medium is nicer?)

Reply:
Personally I think it looks better with `SemiBold` in pills at the top. I think it was made that way deliberately. Both `RoomSpecialCategoyPill` and `RoomStatusPill` now look different, not just one class.

There was also another pill class `RankRangePill` which didn't really fit among these ones, because it has an icon on the left. But it is not used anywhere except in a test, so I didn't bother with it. I think the icon could actually be dropped completely. I don't think it gives any valuable information. *(I hope to see this pill added to the room soon as it does look useful)*

Reply:
> Add a protected virtual FontUsage Font { get; } to OnlinePlayPill and supply that to creationParameters on the TextFlowContainer itself in OnlinePlayPill's BDL

I've applied this solution.